### ISSUE TITLE
Support scheduled/recurring transactions

### ISSUE THREAD

**Issue by justbudget** [justbudget/justbudget#26]
I should be able to add transactions on future dates. I should also be able to add a transaction that repeats weekly/monthly/annually etc.

**Comment by InsouciantQualms** [comment]
This may be a more advanced feature and in a later release, but full support for the [recurrence rule spec](https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html) would be wonderful!

**Comment by justbudget** [comment]
/u/bpaulien had a good intermediate solution idea for this as well, having a "duplicate transaction" button. This would make it really easy to re-create a previous transaction.

See: https://www.reddit.com/r/justbudget/comments/v12vil/comment/ieg2gcc/?utm_source=share&utm_medium=web2x&context=3

**Comment by justbudget** [comment]
Just as additional note, on desktop maybe instead of a button it could be a "right click drop-down menu", where "edit" and "delete" could also live.

**Comment by bpaulien** [comment]
One more note/comment/question regarding how the "future" (scheduled/recurring) transactions should behave.  I started trying to implement some type of workaround for this, and put in a temporary filler for one of the incomes that my wife should have in a couple weeks.  Then I promptly forgot about it and came back the next day to see a $600 discrepancy between my JB budget and Actual budget, which I'm running in parallel.  After a couple hours research, checking everything thoroughly and coming up empty (I thought I had extra in JB, so was trying to find out why Actual was missing this money) I finally realized it was the future transaction I had put in.

So my comment is probably don't make future transactions affect the budget as though they had already happened.  And my question is, is that the way we want it to behave?  Or possibly make it user configurable, so if somebody wants to see their future Tx affect the budget they have that choice, but others can opt out?  

I know this is coming up soon, as it's one of the few remaining P1 issues, and is probably the most requested one.  Just wanted to make this point, and see if anybody has other comments on it.

**Comment by Kidglove57** [comment]
Thanks for your thoughtful post. To add my own perspective - if I want a future transaction to effect my running balance I would post it early and then change its status from pending to open. So just to double check, are you saying that you would not want any future dated transaction to effect the running balance - or just those that are scheduled but not yet posted? Or continue to allow future transactions to be posted but keep them marked as pending? Sorry - I may be getting confused! Just wondering if JBs status choice of pending/open/cleared might help here ?
PS so I would argue that transactions marked pending should not effect the account balance in the sidebar (which they don't) or at the top of the account register (which they currently do).

**Comment by bpaulien** [comment]
Ok, that's a good question, and I'm glad you brought up the distinction.  Since I don't know exactly how this will be implemented, I can't completely answer how I'd like it to work.  I guess my original point was more the actual Budget tab.  I saw that there was an extra $600 in my Available to Budget. And I couldn't reconcile it with my numbers in Actual.  (Actual also showed the 600 in future transactions, it just hadn't affected the actual budget categories yet.)

And that brings up another point, which is, I don't mind my pending (as we currently know it) transactions being able to affect the budget, as in subtract from the category, so it looks like the money is spent.  Because really, the money IS spent, I entered it in when I spent it, it just hasn't been cleared in the bank yet.

But if I have a biweekly income event, I don't want the "pending?" scheduled transaction that is waiting to become real in 2 weeks to show up in the budget on the day that it's created (right after my current income happens), I want the next one to show up in the register, (or wherever scheduled transactions will live) but not affect the actual budget page yet.  And at that point, I guess I don't want it to affect the balance on the sidebar either, but if it shows up on the register as scheduled, I'd still want to see the running balance be affected by it.

As far as how it affects the balance of the accounts, are you saying that the sidebar account balance should not include pending?  So in that case if you are reconciled, your total at the top should match the total along the side bar?  I guess that makes sense, but I'd want to be able to see both totals at some point, maybe in a header that shows reconciled balance, and total, unreconciled balance, because sometimes I want to see both.  -- Oh wait, yeah, I guess you see that in the running balance, that shows everything, whether cleared or not.

**Comment by Kidglove57** [comment]
All great points - and yes I agree with you that posted (but marked pending) transactions should effect the budget. I guess they would not have been posted if that was not the intention. Which then brings up the issue of being able to clearly see the total of "on budget" accounts balances that make up the total of the "available in the budget" ....

I need to think further!

**Comment by Rick-Thomas** [comment]
Let me add my thoughts to this conversation...   A future transaction is something that will occur at a given moment in the future, be it a week, a month, a quarter or a year...  Considering this I think that:

1. As @bpaulien says, it shouldn't hit the Budget till the time is correct...   No future transaction should affect the budget...
2. Elaborating on what I understand of @Kidglove57's comment, it could be interesting if, in each account, one could see the future transactions, probably greyed out on top of the current ones, with the possibility of an option to be able to hide them, where one can see how they affect the balance of the account and decide if transfers have to be made into the account before that date to cover the expense...
3. I don't thing future transactions should be involved in the Pending, Cleared or Reconciled Status, since those affect the account/bank relationship, where future transactions have not yet occurred...

**Comment by Kidglove57** [comment]
I totally agree with Rick’s summary. Just to clarify my comments regarding pending/open/cleared were solely in the context of future (scheduled) transactions that the individual has chosen to post early.

**Comment by bpaulien** [comment]
One other comment, which could possibly have gone unsaid, but just to make sure it's known...  I'd like the ability to select a currently existing transaction, and select, or right click, or something, which will give me the options to turn it into a repeating transaction (keeping all the information in all the fields -- except date -- intact), so I don't have to enter all the information again, just to create a new schedule/repeat.

**Comment by Rick-Thomas** [comment]
Totally agree with @bpaulien ´s comment...   It happens a lot when I pay something in installments...   I enter the 1st one today, with all the details, and then all I want to do is convert that same transaction into a recurring transaction to be used for the future installments...
And while I'm on installments, a **VERY** useful addition would be to be able to tell the recurring transaction the amount of installments, so it ends naturally and I don't have to have a legend in the Note indicating the amount of missing installments!!!

**Comment by Kidglove57** [comment]
Whatever the ultimate "shape" of the solution I love the fact that this is P1.

The four blockers for me in committing to JB 100% are:

1) Splits
2) Recurring transactions
3) Transactions in investment (off budget) accounts 
4) Notes or goals in categories
